biorbd_optim/constraints.py

```diff
@@ -50,77 +50,112 @@ def add_constraints(ocp, nlp):
         """
         if nlp["constraints"] is None:
             return
-        for elem in nlp["constraints"]:
-            if elem[1] == Constraint.Instant.START:
-                x = [nlp["X"][0]]
-                u = [nlp["U"][0]]
-            elif elem[1] == Constraint.Instant.MID:
-                if nlp["ns"] % 2 == 0:
-                    raise (ValueError("Number of shooting points must be odd to use MID"))
-                x = [nlp["X"][nlp["ns"] // 2 + 1]]
-                u = [nlp["U"][nlp["ns"] // 2 + 1]]
-            elif elem[1] == Constraint.Instant.INTERMEDIATES:
-                x = nlp["X"][1 : nlp["ns"] - 1]
-                u = nlp["U"][1 : nlp["ns"] - 1]
-            elif elem[1] == Constraint.Instant.END:
-                x = [nlp["X"][nlp["ns"]]]
-                u = []
-            elif elem[1] == Constraint.Instant.ALL:
-                x = nlp["X"]
-                u = nlp["U"]
+        for constraint in nlp["constraints"]:
+            x, u, last_node = Constraint.__get_instant(nlp, constraint)
+            type = constraint["type"]
+            del constraint["instant"], constraint["type"]
```

Comment:
What is that for?

Reply:
a function who has a dict in a parameter has to have all dict keys in these parameters, so i delete type and instant keys.